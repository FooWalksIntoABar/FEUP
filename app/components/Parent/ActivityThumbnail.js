import React from 'react'
import { View, Image, StyleSheet } from 'react-native'
import { PropTypes } from 'prop-types'

export class SortableListItemThumbnail extends React.Component {
  styles = StyleSheet.create({
    view: {
      height: 50,
      width: 50,
      borderRadius: 25,
      marginRight: 15,
      justifyContent: 'center',
      alignItems: 'center'
    },
    image: {
      width: '60%',
      height: '60%'
    }
  })

  render () {
    return (
      <View style={[this.styles.view, { backgroundColor: this.props.color }]}>
        <Image source={this.props.source} style={this.styles.image} resizeMode={'contain'} />
      </View>
    )
  }
}

SortableListItemThumbnail.propTypes = {
  color: PropTypes.string.isRequired,
  source: PropTypes.number.isRequired
}
