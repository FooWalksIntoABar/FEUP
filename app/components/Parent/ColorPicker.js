import React, { Component } from 'react'
import { Picker, Item } from 'native-base'
import PropTypes from 'prop-types'

export class ColorPicker extends Component {
  onValueChange = value => {
    this.props.onColorChange(this.props.colors[value].code)
  }

  render () {
    const pickerItems = this.props.colors.map((color, index) => (<Picker.Item label={color.name} value={index} key={index} />))
    return (
      <Item picker style={{ borderColor: this.props.color }}>
        <Picker
          mode='dropdown'
          style={{ width: 120 }}
          selectedValue={this.props.colors.findIndex(color => color.code === this.props.color)}
          onValueChange={this.onValueChange}
        >
          {pickerItems}
        </Picker>
      </Item>
    )
  }
}

ColorPicker.propTypes = {
  color: PropTypes.string.isRequired,
  colors: PropTypes.array.isRequired,
  onColorChange: PropTypes.func.isRequired
}
