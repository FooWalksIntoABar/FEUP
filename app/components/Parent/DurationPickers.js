import React, { Component } from 'react'
import { View } from 'react-native'
import { Item, Label, Input, Button, Text } from 'native-base'
import PropTypes from 'prop-types'

export class DurationPickers extends Component {
  constructor (props) {
    super(props)
    this.state = {
      detailedView: false
    }
  }

  toggleDetailedView = () => {
    this.setState({ detailedView: !this.state.detailedView })
  }

  changeGoalValue = (value) => {
    this.props.onDurationChange({ goal: value.replace(/[^\d.-]/g, ''), interpolate: this.state.detailedView })
  }

  changeMaxValue = (value) => {
    this.props.onDurationChange({ max: value.replace(/[^\d.-]/g, ''), interpolate: this.state.detailedView })
  }

  changeMinValue = (value) => {
    this.props.onDurationChange({ min: value.replace(/[^\d.-]/g, ''), interpolate: this.state.detailedView })
  }

  render () {
    return (
      <View>
        <Item stackedLabel>
          <Label>{this.state.detailedView ? 'Duração ideal (minutos)' : 'Duração (minutos)'}</Label>
          <Input keyboardType='numeric' textContentType='telephoneNumber' value={this.props.time.goal} onChangeText={this.changeGoalValue} />
        </Item>
        {this.state.detailedView && <Item stackedLabel>
          <Label>Duração Minima (minutos)</Label>
          <Input keyboardType='numeric' textContentType='telephoneNumber' value={this.props.time.min} onChangeText={this.changeMinValue} />
        </Item>}
        {this.state.detailedView && <Item stackedLabel>
          <Label>Duração Máxima (minutos)</Label>
          <Input keyboardType='numeric' textContentType='telephoneNumber' value={this.props.time.max} onChangeText={this.changeMaxValue} />
        </Item>}
        <View style={{ flexDirection: 'row', justifyContent: 'flex-end' }}>
          <Button transparent onPress={this.toggleDetailedView}>
            <Text style={{ color: this.props.color }}>{this.state.detailedView ? 'Menos opções de duração' : 'Mais opções de duração'}</Text>
          </Button>
        </View>
      </View>
    )
  }
}

DurationPickers.propTypes = {
  onDurationChange: PropTypes.func,
  time: PropTypes.object.isRequired,
  color: PropTypes.string.isRequired
}
