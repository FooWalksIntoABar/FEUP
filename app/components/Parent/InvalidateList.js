import React from 'react'
import { List, ListItem, Left, Text, Right, Icon, Button } from 'native-base'
import { View, Image } from 'react-native'
import { SortableListItemThumbnail } from './SortableListItemThumbnail'
import PropTypes from 'prop-types'
import { getSource } from '../../helpers/GetSource'
import { red } from '../../styles/Colors'
import images from '../../assets/images/images'

export class InvalidateList extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      activities: this.props.activities
    }
    if (this.props.activities == null) {
      this.state.activities = []
    }
  }

  formatTime = time => {
    if (time <= 0) {
      return '0:00'
    }

    let minutes = Math.floor(time / 60)

    let seconds = Math.floor(time % 60)
    if (seconds < 10) seconds = '0' + seconds

    return `${minutes}:${seconds}`
  }

  render () {
    console.log(this.props.activities)
    const activities = this.props.activities.map((activity, index) => (
      <ListItem button onPress={this.onItemPress} key={index}>
        <Left>
          <SortableListItemThumbnail isPhoto={activity.photo !== null} color={activity.color} source={getSource(activity)} />
          <View style={{ flex: 1 }}>
            <Text style={{ alignSelf: 'flex-start' }}>{activity.title}</Text>
            <View style={{ flexDirection: 'row', alignItems: 'center' }}>
              <Image source={images.ui.star} style={[{ height: 12, width: 12, margin: 2 }, activity.reward < 1 && { tintColor: 'lightgrey' }]} />
              <Image source={images.ui.star} style={[{ height: 12, width: 12, margin: 2 }, activity.reward < 2 && { tintColor: 'lightgrey' }]} />
              <Image source={images.ui.star} style={[{ height: 12, width: 12, margin: 2 }, activity.reward < 3 && { tintColor: 'lightgrey' }]} />
              <Text style={{ alignSelf: 'flex-start' }} note numberOfLines={1}>{`- ${activity.reward ? 'Completada' : 'Falhada'} em ${this.formatTime(activity.elapsedTime)}`}</Text>
            </View>
          </View>
        </Left>
        <Right>
          <Button rounded onPress={() => this.props.onInvalidatePress(index)} style={{ backgroundColor: red }}>
            <Icon name={'md-close'} />
          </Button>
        </Right>
      </ListItem>))

    return (
      <List style={{ width: '100%' }}>
        {activities}
      </List>
    )
  }
}

InvalidateList.propTypes = {
  activities: PropTypes.array.isRequired,
  onInvalidatePress: PropTypes.func.isRequired
}
