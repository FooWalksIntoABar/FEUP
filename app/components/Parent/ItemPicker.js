import React, { Component } from 'react'
import { Picker } from 'native-base'
import PropTypes from 'prop-types'

export class ItemPicker extends Component {
  render () {
    const pickerItems = this.props.items.map((item, index) => (<Picker.Item label={item} value={index} key={index} />))
    return (
      <Picker
        mode='dropdown'
        style={{ width: '100%' }}
        selectedValue={this.props.items.findIndex(item => item === this.props.selected)}
        onValueChange={this.props.onValueChange}
      >
        {pickerItems}
      </Picker>
    )
  }
}

ItemPicker.propTypes = {
  items: PropTypes.array.isRequired,
  selected: PropTypes.any,
  onValueChange: PropTypes.func.isRequired
}
