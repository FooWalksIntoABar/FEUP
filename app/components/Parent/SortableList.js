import React from 'react'
import { List } from 'native-base'
import { SortableListItem } from './SortableListItem'
import PropTypes from 'prop-types'

export class SortableList extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      items: this.props.items
    }

    this.moveItemUp = this.moveItemUp.bind(this)
  }

  moveItemUp = (i) => {
    if (i === 0) return
    this.setState({ items: this.state.items.map((element, index) => {
      if (index === i - 1) return this.state.items[i]
      else if (index === i) return this.state.items[i - 1]
      else return element
    }) })
  }

  render () {
    const items = this.state.items.map((item, index) => (<SortableListItem item={item} index={index} key={index} moveItemUp={this.moveItemUp} onItemPress={this.props.onItemPress} color={this.props.color} />))

    return (
      <List style={{ width: '100%' }}>
        {items}
      </List>
    )
  }
}

SortableList.propTypes = {
  items: PropTypes.array.isRequired,
  onItemPress: PropTypes.func.isRequired,
  color: PropTypes.string
}
