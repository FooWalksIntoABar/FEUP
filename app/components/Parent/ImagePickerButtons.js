import React, { Component } from 'react'
import { ImagePicker, Permissions } from 'expo'
import { Label, Button, Text, Item } from 'native-base'
import { View, StyleSheet, Image } from 'react-native'
import PropTypes from 'prop-types'

import Images, { availableImages } from '../../assets/images/images'
import { ItemPicker } from './ItemPicker'

export class ImagePickerButtons extends Component {
  constructor (props) {
    super(props)
    this.state = {
      photoView: true
    }
  }

  togglePhotoView = () => {
    this.setState({ photoView: !this.state.photoView })
  }

  imagePicker = async (camera) => {
    const options = { allowsEditing: true, aspect: [5, 3] }
    let result

    if (camera) {
      await Permissions.askAsync(Permissions.CAMERA)
      result = await ImagePicker.launchCameraAsync({ ...options, mediaTypes: 'Images' })
    } else {
      await Permissions.askAsync(Permissions.CAMERA_ROLL)
      result = await ImagePicker.launchImageLibraryAsync(options)
    }

    if (!result.cancelled) {
      this.props.onPhotoChange(result.uri)
    }
  }

  onImageChanged = (index) => {
    this.props.onImageChange(availableImages.map(item => item.image)[index])
  }

  getSelectedImageName = () => {
    const image = availableImages.find(item => item.image === this.props.image)
    if (image) return image.name
    else return undefined
  }

  renderOption = () => {
    if (this.state.photoView) {
      return (
        <View style={styles.itemContainer}>
          {this.props.photo && <View style={styles.imageContainer}>
            <Image style={styles.photoPreview} source={this.props.photo} resizeMode={'contain'} />
          </View>}
          <View style={styles.imageButtonContainer}>
            <Button style={[styles.imageButton, { backgroundColor: this.props.color }]} onPress={() => this.imagePicker(true)}>
              <Text>Camara</Text>
            </Button>
            <Button style={[styles.imageButton, { backgroundColor: this.props.color }]} onPress={() => this.imagePicker(false)}>
              <Text>Galeria</Text>
            </Button>
          </View>
        </View>)
    } else {
      return (
        <View style={styles.itemContainer}>
          {this.props.image && <View style={styles.imageContainer}>
            <Image style={styles.imagePreview} source={Images[this.props.image]} resizeMode={'contain'} />
          </View>}
          <Item style={{ borderColor: 'transparent' }}>
            <ItemPicker items={availableImages.map(item => item.name)} selected={this.getSelectedImageName()} onValueChange={this.onImageChanged} />
          </Item>
        </View>
      )
    }
  }

  render () {
    return (
      <View>
        <Item stackedLabel>
          <Label>Imagem</Label>
          {this.renderOption()}
        </Item>
        <View style={{ flexDirection: 'row', justifyContent: 'flex-end' }}>
          <Button transparent onPress={this.togglePhotoView}>
            <Text style={{ color: this.props.color }}>{this.state.photoView ? 'Escolher imagem' : 'Escolher fotografia'}</Text>
          </Button>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  itemContainer: {
    flexDirection: 'column',
    width: '100%'
  },
  imageButtonContainer: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'flex-start'
  },
  imageButton: {
    flex: 1,
    marginRight: 15,
    marginVertical: 15,
    justifyContent: 'center'
  },
  photoPreview: {
    width: '100%',
    aspectRatio: 1.666
  },
  imagePreview: {
    width: 100,
    height: 100
  },
  imageContainer: {
    flexGrow: 1,
    paddingRight: 15,
    marginTop: 15,
    alignItems: 'center'
  }
})

ImagePickerButtons.propTypes = {
  color: PropTypes.string.isRequired,
  onPhotoChange: PropTypes.func.isRequired,
  onImageChange: PropTypes.func.isRequired,
  photo: PropTypes.object,
  image: PropTypes.string
}
