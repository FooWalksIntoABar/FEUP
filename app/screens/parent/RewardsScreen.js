import React from 'react'
import { Content } from 'native-base'
import { SortableList } from '../../components/Parent/SortableList'

export default class RewardsScreen extends React.Component {
  render () {
    return (
      <Content>
        <SortableList items={rewards} onItemPress={null} />
      </Content>
    )
  }
}

const rewards = [
  {
    title: 'Ir Ã  piscina',
    photo: 'https://upload.wikimedia.org/wikipedia/commons/e/e3/Magnificent_CME_Erupts_on_the_Sun_-_August_31.jpg'
  },
  {
    title: 'Brincar com Legos',
    photo: 'https://upload.wikimedia.org/wikipedia/commons/e/e3/Magnificent_CME_Erupts_on_the_Sun_-_August_31.jpg'
  },
  {
    title: 'Jogar PlayStation',
    photo: 'https://upload.wikimedia.org/wikipedia/commons/e/e3/Magnificent_CME_Erupts_on_the_Sun_-_August_31.jpg'
  }
]
