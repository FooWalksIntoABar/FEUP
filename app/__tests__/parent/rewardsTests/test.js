import React from 'react'
import { SortableList } from '../../../components/Parent/SortableList'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import RewardsScreen from '../../../screens/parent/RewardsScreen'

configure({ adapter: new Adapter() })

describe('RewardsScreen snapshot', () => {
  it('renders RewardsScreen correctly', async () => {
    const wrapper = shallow(<RewardsScreen
      navigation={{ navigate: jest.fn(), pop: jest.fn() }} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find(SortableList).at(0).props().onItemPress()
    wrapper.unmount()
  })
})
