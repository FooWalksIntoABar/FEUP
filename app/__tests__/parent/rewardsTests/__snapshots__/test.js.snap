// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewardsScreen snapshot renders RewardsScreen correctly 1`] = `
<Styled(Content)>
  <SortableList
    items={
      Array [
        Object {
          "photo": "https://upload.wikimedia.org/wikipedia/commons/e/e3/Magnificent_CME_Erupts_on_the_Sun_-_August_31.jpg",
          "title": "Ir Ã  piscina",
        },
        Object {
          "photo": "https://upload.wikimedia.org/wikipedia/commons/e/e3/Magnificent_CME_Erupts_on_the_Sun_-_August_31.jpg",
          "title": "Brincar com Legos",
        },
        Object {
          "photo": "https://upload.wikimedia.org/wikipedia/commons/e/e3/Magnificent_CME_Erupts_on_the_Sun_-_August_31.jpg",
          "title": "Jogar PlayStation",
        },
      ]
    }
    onItemPress={[Function]}
  />
</Styled(Content)>
`;
